### ISSUE TITLE
Save Autoconfig overwrites device #0 instead of currently selected device

### ISSUE THREAD

**Issue by jvook** [libretro/RetroArch#3578]
If you have two controllers plugged in and try to save an autoconfig profile on the second controller, that config will be written to the first recognized device regardless of whether it is the currently-selected device.

This results in nonsensical autoconfig profiles for the first-recognized device after selecting to Save Autoconfig on the intended device.

**Comment by fr500** [comment]
I don't really get this issue. 
Saving just saves, it doesn't apply settings.

**Comment by jvook** [comment]
It saves to the first device's autoconfig.  So say you have a ps2 controller as first device, and x360 as second device.  If you config your x360 controller and save autoconfig, it writes your 360 binds to the ps2 autoconfig.

**Comment by jvook** [comment]
@fr500 I'm not sure how to address unconfirmed label.  here is a video of the bug: https://youtu.be/jeG649ScqkI

**Comment by lasers** [comment]
Hi. I think I found something.

**Round One:** *Creating PS3 config with Xbox controller.*

Use PS3 controller. Go to `Settings`... `Input User 1 Binds`... `User 1 Device Index`.
Switch from PS3 to Xbox. Dead PS3 controller.

Time to set up Xbox controller. Use Xbox controller.
Go to `User 1 Bind All`. Press Dpad Up all the way. Dead Xbox controller (almost).

Use keyboard. Go to `User 1 Save Autoconfig` -- Enter.
Config generated for PS3 when we made one for Xbox controller.

Inside`'Sony PLAYSTATION(R)3 Controller.cfg'`

```
input_driver = "udev"
input_device = "Sony PLAYSTATION(R)3 Controller"
input_vendor_id = "1356"
input_product_id = "616"
input_select_btn = "h0up"
input_start_btn = "h0up"
input_up_btn = "h0up"
input_down_btn = "h0up"
input_left_btn = "h0up"
input_right_btn = "h0up"
input_a_btn = "h0up"
input_x_btn = "h0up"
input_l_btn = "h0up"
input_r_btn = "h0up"
input_l2_btn = "h0up"
input_r2_btn = "h0up"
input_l3_btn = "h0up"
input_r3_btn = "h0up"
input_l_x_plus_btn = "h0up"
input_l_x_minus_btn = "h0up"
input_l_y_plus_btn = "h0up"
input_l_y_minus_btn = "h0up"
input_r_x_plus_btn = "h0up"
input_r_x_minus_btn = "h0up"
input_r_y_plus_btn = "h0up"
input_r_y_minus_btn = "h0up"
```
Still no Xbox config.

**Round two:** *Creating Xbox config with PS3 controller.*

Xbox should be default and in charge by now. Use Xbox controller... barely. Up. Up. Up.

Use keyboard. Go back out. Go to `Input 2 Binds`... `User 2 Device Index`.
NOTE: `User 2 Device Index` is already set for Xbox. Switch over to PS3.

Use keyboard. Go to `User 2 Bind All`. Use PS3 controller. Press Dpad Down all the way.

Go to `User 2 Save Autoconfig` -- Enter.
Config generated for Xbox when we made one for PS3 controller.

Inside `'Microsoft X-Box 360 pad.cfg'`
```
input_driver = "udev"
input_device = "Microsoft X-Box 360 pad"
input_vendor_id = "1118"
input_product_id = "654"
input_b_btn = "6"
input_y_btn = "6"
input_select_btn = "6"
input_start_btn = "6"
input_up_btn = "6"
input_down_btn = "6"
input_left_btn = "6"
input_right_btn = "6"
input_a_btn = "6"
input_x_btn = "6"
input_l_btn = "6"
input_r_btn = "6"
input_l2_btn = "6"
input_r2_btn = "6"
input_l3_btn = "6"
input_r3_btn = "6"
input_l_x_plus_btn = "6"
input_l_x_minus_btn = "6"
input_l_y_plus_btn = "6"
input_l_y_minus_btn = "6"
input_r_x_plus_btn = "6"
input_r_x_minus_btn = "6"
input_r_y_plus_btn = "6"
input_r_y_minus_btn = "6"
```
Still no PS3 config.

**Round three:** *Dead controller*
Use keyboard. Go to `User 1 Binds`... Switch `User 1 Device` from Xbox to PS3.
Use PS3 controller. The controller is non-responsive for me at this point.

**tl;dr:** *Changing `User 0 Device Index`, `User 0 Bind All`, and `User 0 Save Autoconfig` all at once configures the new settings for old controller (for me, at least).*

**Comment by jvook** [comment]
To avoid this issue you just need to make sure that only the intended device is plugged in when generating the autoconfig

**Comment by LetsDoThisGitThing** [comment]
I confirm too the issue... I have 3 gamepads, and this issue made me crazy to understand why it wasn't saving the configuration.

What jvook suggested, is problematic for bluetooth gamepads, because them are listed even if not connected... so in this case to have only one device, you have also to remove the pairing of all the not required gamapads.

**Comment by jvook** [comment]
I found this in configuration.c 
<html>
<body>


/**
--
3982 | * config_save_autoconf_profile:
3983 | * @device_name       : Input device name
3984 | * @user              : Controller number to save
3985 | * Writes a controller autoconf file to disk.
3986 | **/


</body>
</html>
So I think it is not getting the user variale properly
<html>
<body>


/**
--
3982 | * config_save_autoconf_profile:
3983 | * @device_name       : Input device name
3984 | * @user              : Controller number to save
3985 | * Writes a controller autoconf file to disk.
3986 | **/


</body>
</html>
